legion/engine/physics/physx/physx_integration_helpers.cpp

```diff
@@ -17,21 +17,29 @@ namespace legion::physics
         pxTransform.q = { rot.x,rot.y,rot.z,rot.w };
     }
 
-    void calculateGlobalAndLocalTransforms(physx::PxTransform& outLocalTransform, physx::PxTransform& outGlobalTransform, const ColliderData& collider, ecs::entity ent)
+    void calculateLocalColliderTransform(physx::PxTransform& outLocalTransform, const ColliderData& collider)
     {
-        const math::vec3& pos = *ent.get_component<position>();
-        const math::quat& rot = *ent.get_component<rotation>();
-
         const math::vec3& localOffset = collider.getOffset();
         const math::quat& localRot = collider.getRotationOffset();
 
-        toPhysxTransform(outGlobalTransform, pos, rot);
         toPhysxTransform(outLocalTransform, localOffset, localRot);
     }
 
+    void calculateGlobalAndLocalTransforms(physx::PxTransform& outLocalTransform, physx::PxTransform& outGlobalTransform, const ColliderData& collider, ecs::entity ent)
+    {
+        transform trans = ent.get_component<transform>();
+        const math::mat4& globalTransform = trans.to_world_matrix();
+
+        math::vec3 pos; math::quat rot; math::vec3 tempScale;
+        math::decompose(globalTransform, tempScale, rot, pos);
+       
+        calculateLocalColliderTransform(outLocalTransform, collider);
+        toPhysxTransform(outGlobalTransform, pos, rot);
```

Comment:
still kindah interesting that physx doesn't require scale here, but i looked it up and ig this makes sense XD

Reply:
also actually explains the collider scaling bug diviner has ;D